### PR TITLE
Feature/convert splitted zip to parquet

### PR DESCRIPTION
**Pull request by LidorDublin**

### CODE REVIEW

File: tools/files/concat_csv_files.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        dst: Destination directory
+        dry: Whether running dry or not
+
+    Returns:
+        List of files extracted from the given files list
+    """
+
+    extracted_files = []
+    results = list(futures.ProcessPoolExecutor().map(unzip_file, files, repeat(dst), repeat(dry)))  # we have to include the list(...) so it will wait for all the as async function calls to finish
+    for _ in results:
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: _ has meaning here

File: tools/files/concat_csv_files.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+    files_extracted = []
+    suffix = file.suffix.lower()
+    if suffix == '.zip':
+        with zipfile.ZipFile(file, 'r') as zip_ref:
+            for unzipped_file in zip_ref.filelist:
+                files_extracted.append(dst / unzipped_file.filename)
+                print(f"Unzipping '{file}': {unzipped_file.filename} -> {dst / unzipped_file.filename}")
+
+            if not dry:
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: Move before the loop

File: tools/files/concat_csv_files.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    extracted_files = []
+    results = list(futures.ProcessPoolExecutor().map(unzip_file, files, repeat(dst), repeat(dry)))  # we have to include the list(...) so it will wait for all the as async function calls to finish
+    for _ in results:
+        extracted_files.extend(_)
+
+    return extracted_files
+
+
+def concat_files(files: list):
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: add return type annotations if you will

File: tools/files/concat_csv_files.py [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+    name = search('_S(\d+)_cbe(DDV\w+)', files[0].stem)
+    if name is None:
+        raise ValueError
+
+    name = name.groups()
+    concatenated_file = files[0].parent / f"{CONSTANT_PART_OF_NAME}{name[0]}.{name[1]}{files[0].suffix}"
+
+    for file in files:
+        system(f'cat {file} >> {concatenated_file} && rm {file}')
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: It's better to import os and use os.system for this cmd

File: tools/files/concat_csv_files.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+    return concatenated_file
+
+
+def main():
+    files, dst, run_dry = parse_args()
+    if not files:
+        raise TypeError("Expected list of files (-f or --files) as argument.")
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: I think this can be handled in parse args

File: tools/files/concat_csv_files.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+
+    if not dst:
+        raise TypeError("Expected destination directory (-d or --dst) as argument.")
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: same

File: tools/files/concat_csv_files.py (hunk 4 above)
Comment (LidorDublin) [reply tools/files/concat_csv_files.py]: mind explaining why it's better?

File: tools/files/concat_csv_files.py [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        files: List of files to concatenate
+
+    Returns:
+        The size of the new file
+    """
+
+    if not files:
+        raise ValueError
+
+    name = search(r'_S(\d+)_cbe(DDV\w+)', files[0].stem)
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: add comment to explain regex

File: tools/files/concat_csv_files.py [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    if name is None:
+        raise ValueError
+
+    name = name.groups()
+    concatenated_file = files[0].parent / f"{CONSTANT_PART_OF_NAME}{name[0]}.{name[1]}{files[0].suffix}"
+
+    for file in files:
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: file -> splitted_file or something better

File: tools/files/concat_csv_files.py [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+
+    # Otherwise the async-called functions may overlap with the mkdir and it will crush the program
+    if run_dry:
+        print("Dry run:")
+        unzip_files(files, dst, True)
+    else:
+        dst.mkdir(exist_ok=True)
+        # result_file_name = search('PRM.EDTOUT.DGMPAGYA.P\d+\.(ACC|DDV\w+)', files[0])
+        # print(result_file_name)
+        parquet_converter.convert_files([concat_files(unzip_files(files, dst))], True)
```
Comment (hodisr) [inline tools/files/concat_csv_files.py]: Dont convert here, and zip the files after concatenation

**Pull request closed without merging**